apps/frontend/src/libs/components/table-pagination/table-pagination.tsx

```diff
@@ -0,0 +1,133 @@
+import { useCallback, useMemo } from "~/libs/hooks/hooks.js";
+
+import { ChangePageButton } from "./libs/components/components.js";
+import styles from "./styles.module.css";
+
+const options = [
+	{ label: "10", value: 10 },
+	{ label: "25", value: 25 },
+	{ label: "50", value: 50 },
+];
+
+type Properties = {
+	onPageChange: (page: number) => void;
+	onPageSizeChange: (pageSize: number) => void;
+	page: number;
+	pageSize: number;
+	totalItems: number;
+};
+
+// TODO: replace this select with app select
+// ===>
+type MockSelectProperties = {
+	onChange: (value: number) => void;
+	options: { label: string; value: number }[];
+};
+
+const MockSelect = ({
+	onChange,
+	options,
+}: MockSelectProperties): JSX.Element => {
+	const handleOnChange = useCallback(
+		(event: React.ChangeEvent<HTMLSelectElement>) => {
+			onChange(Number(event.target.value));
+		},
+		[onChange],
+	);
+
+	return (
+		<select onChange={handleOnChange}>
+			{options.map(({ label, value }) => (
+				<option key={label} value={value}>
+					{label}
+				</option>
+			))}
+		</select>
+	);
+};
+// <===
+
+const FIRST_PAGE = 1;
+const PAGE_INCREMENT = 1;
+
+const TablePagination = ({
+	onPageChange,
+	onPageSizeChange,
+	page,
+	pageSize,
+	totalItems,
+}: Properties): JSX.Element => {
+	const totalPages = useMemo(
+		() => Math.ceil(totalItems / pageSize),
+		[totalItems, pageSize],
+	);
+
+	const hasNextPage = useMemo(() => page < totalPages, [page, totalPages]);
+	const hasPreviousPage = useMemo(() => page > FIRST_PAGE, [page]);
```

Comment:
I think we don't need use memo here, too small check to allocate additional memory for this

Reply:
So you suggest to replace `useMemo` with `useCallback`? If yes, what benefits this bring? If no, one of the efficient 
solution may be to move this functions out of the component (with aproppriate parameters) and then use them:
```ts
const hasNextPage = (page: number, totalPages: number): boolean => page < totalPages;
const hasPreviousPage = ...;

const TablePagination = (...) => {
  ...
  isDisabled={!hasNextPage(page, totalPages)}
  ...
}
```

Reply:
I suggest to write it without any hooks (if there is no restriction in linter) `const hasNextPage = page < totalPages`

Reply:
Oh, but without useMemo hook our component won't be rerendered whenever `page` or `totalPages` change, if I'm not wrong

Reply:
Leave it as it is, because it does not work without useMemo hook